todo/tests.py

```diff
@@ -1,6 +1,141 @@
+
 
 # Create your tests here.
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+        django_get_or_create = ('username',)
+
+    username = 'tester'
+    email = 'tester@jtester.com'
+    password = 'top_secret'
+
+
+class TaskFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = Task
+
+    title = factory.Sequence(lambda n: 'Title_{}'.format(n))
+    text = factory.Sequence(lambda n: 'some_text_{}'.format(n))
+    status = factory.Iterator([Task.TODO, Task.IN_PROGRESS, Task.DONE, Task.TO_DELETE])
+    author = factory.SubFactory(UserFactory, username='tester')
+
+
+class TodoViewsTests(TestCase):
+
+    def setUp(self):
+        self.f = RequestFactory()
+        self.user = UserFactory()
+        self.task0, self.task1, self.task2, self.task3 = [TaskFactory() for i in range(4)]
+
+    def tearDown(self):
+        TaskFactory.reset_sequence(0)
+
+    def test_should_display_correct_list(self):
+        request = self.f.get(reverse('tasklist'))
+        request.user = self.user
+        view = TaskListView.as_view()
+        response = view(request)
+        response.render()
+        self.assertTrue(response.context_data["todo"])
+        self.assertTrue(response.context_data["in_progress"])
+        self.assertTrue(response.context_data["done"])
+        self.assertEqual(len(response.context_data["todo"]), 1)
+        self.assertEqual(len(response.context_data["in_progress"]), 1)
+        self.assertEqual(len(response.context_data["done"]), 1)
+        with self.assertRaises(KeyError):
+            response.context_data["to_delete"]
+
+    def test_detail_view_should_display_correct_record(self):
+        request = self.f.get(reverse('taskdetail', kwargs={'pk': "1"}))
+        request.user = self.user
+        view = TaskDetailView.as_view()
+        kwargs = {'pk': "1"}
+        response = view(request, **kwargs)
+        response.render()
+        self.assertTrue(response.context_data['object'])
+        self.assertEqual(response.context_data['object'].text, "some_text_0")
+
+    def test_update_view_should_update_correct_record(self):
+        request = self.f.get(reverse('updatestatus', kwargs={"pk": "1"}))
+        request.user = self.user
+        form_data = {"status": "in_progress"}
+        request.form = TaskUpdateForm(instance=self.task0, data=form_data)
+        self.assertTrue(request.form.is_valid())
+
+        request.form.save()
+        kwargs = {"pk": "1"}
+        view = UpdateStatusView.as_view()
+        response = view(request, **kwargs)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.task0.status, "in_progress")
+
+    def test_task_form_view_should_have_correct_initial_values(self):
+        request = self.f.post(reverse('taskform', kwargs={"status": "todo"}))
+        request.user = self.user
+        form_data = {"title": "next_title", "text": "more text"}
+        request.form = TaskForm(data=form_data)
```

Comment:
Why are you putting form in request ? View is handling forms itself, don't touch it